pkg/worker/worker.go

```diff
@@ -66,22 +66,25 @@ type Worker struct {
 }
 
 type ContainerInstance struct {
-	Id           string
-	StubId       string
-	BundlePath   string
-	Overlay      *common.ContainerOverlay
-	Spec         *specs.Spec
-	Err          error
-	ExitCode     int
-	Port         int
-	OutputWriter *common.OutputWriter
-	LogBuffer    *common.LogBuffer
+	Id            string
+	StubId        string
+	BundlePath    string
+	Overlay       *common.ContainerOverlay
+	Spec          *specs.Spec
+	Err           error
+	ExitCode      int
+	Port          int
+	OutputWriter  *common.OutputWriter
+	LogBuffer     *common.LogBuffer
+	Request       *types.ContainerRequest
+	ImageMetadata *common.ImageMetadata
 }
 
 type ContainerOptions struct {
-	BundlePath  string
-	BindPort    int
-	InitialSpec *specs.Spec
+	BundlePath    string
+	BindPort      int
+	InitialSpec   *specs.Spec
+	ImageMetadata *common.ImageMetadata
```

Comment:
ImageMetadata is used by getContainerEnvironment() to copy the image metadata vars to the runc spec process env vars.